utils/middlewares/hsl-jwt-validation-middleware.ts

```diff
@@ -0,0 +1,136 @@
+  IResponseErrorForbiddenNotAuthorized,
+  IResponseSuccessJson,
+  ResponseErrorForbiddenNotAuthorized,
+  ResponseSuccessJson,
+  ResponseErrorInternal,
+  IResponseErrorInternal
+} from "@pagopa/ts-commons/lib/responses";
+
+import { readableReportSimplified } from "@pagopa/ts-commons/lib/reporters";
+import { AuthBearer } from "../../generated/definitions/external/AuthBearer";
+import { IConfig } from "../config";
+import { getHubSpidLoginClient } from "../../clients/hubSpidLogin";
+import { IntrospectSuccessResponse } from "../../generated/definitions/hub-spid-login/IntrospectSuccessResponse";
+
+export interface IHslJwtPayloadExtended extends jwt.JwtPayload {
+  readonly name: string;
+  readonly family_name: string;
+  readonly fiscal_number: FiscalCode;
+}
+
+type IjwtIntrospectionCall = (
+  token: NonEmptyString,
+  config: IConfig
+) => TE.TaskEither<
+  IResponseErrorInternal | IResponseErrorForbiddenNotAuthorized,
+  IResponseSuccessJson<IntrospectSuccessResponse>
+>;
+export const introspectionCall: IjwtIntrospectionCall = (token, config) =>
+  pipe(
+    TE.tryCatch(
+      () =>
+        getHubSpidLoginClient(
+          config.HUB_SPID_LOGIN_CLIENT_BASE_URL
+        ).introspectToken({
+          body: {
+            token
+          }
+        }),
+      flow(E.toError, () =>
+        ResponseErrorInternal(
+          `Something went wrong while calling the introspection endpoint`
+        )
+      )
+    ),
+    TE.chain(
+      flow(
+        TE.fromEither,
+        TE.mapLeft(errors =>
+          ResponseErrorInternal(readableReportSimplified(errors))
+        ),
+        TE.chainW(response => {
+          switch (response.status) {
+            case 200:
+              return response.value.active
+                ? TE.right(ResponseSuccessJson(response.value))
+                : TE.left<
+                    | IResponseErrorForbiddenNotAuthorized
+                    | IResponseErrorInternal
+                  >(ResponseErrorForbiddenNotAuthorized);
+            case 403:
+              return TE.left<
+                IResponseErrorForbiddenNotAuthorized | IResponseErrorInternal
+              >(ResponseErrorForbiddenNotAuthorized);
+            default:
+              return TE.left(ResponseErrorInternal(`Something went wrong`));
+          }
+        })
+      )
+    )
+  );
+
+export type HslJWTValid = (
+  token: NonEmptyString
+) => TE.TaskEither<Error, IHslJwtPayloadExtended>;
+
+export const hslJwtValidation = (
+  token: NonEmptyString,
+  config: IConfig
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+): HslJWTValid => () =>
+  pipe(
+    getValidateJWT(
+      config.HUB_SPID_LOGIN_JWT_ISSUER,
+      config.HUB_SPID_LOGIN_JWT_PUB_KEY
+    )(token),
+    TE.chain(jwtDecoded =>
+      pipe(
+        introspectionCall(token, config),
+        TE.fold(
+          _ => TE.left(new Error("Something went wrong")),
+          // active is always true if we are in this rail

```

Comment:
Why? If the token will be invalidated `active=false`

Reply:
If the token after introspection is not active, it returns 200 and active = false. Do you mean to change the error message?

Reply:
The error message it's ok. But if the value of `active` is `false` the handler should return `unauthorized`. This implementation seems to allow invalidated tokens. Can we add a test for this scenario please?

Reply:
based on the [current implementation](https://github.com/pagopa/hub-spid-login-ms/blob/7e029ea0d8d27bf80fe1b1214974c3048bb9674f/src/handlers/token.ts#L124) of the introspect endpoint we always receive `active: true` when the response has status code of 200. however if we want to return unauthorized on this middleware we can filter the `IResponse` returned by `introspectionCall` function.

Reply:
My bad, if `active=false` with status code `200` the response is `ResponseErrorForbiddenNotAuthorized`. But the implementation seems to return always `true` with that http status code. So the type for the generated client is wrong in [here](https://github.com/pagopa/io-web-profile-backend/pull/8/files#diff-e13b8fe59f2873727f8133419563d711b0b3d031ef9af3b7b529e3557aaafe39R48)
Maybe should be like
```yaml
IntrospectSuccessResponse:
  type: object
  properties:
    active:
      type: boolean
      enum:
        - true
  required:
    - active
  example:
    active: true
IntrospectForbiddenResponse:
  type: object
  properties:
    active:
      type: boolean
      enum:
        - false
  required:
    - active
  example:
    active: false
```